src/filemenu.cpp

```diff
@@ -226,7 +226,7 @@ FileMenu::FileMenu(Fm::FileInfoList files, std::shared_ptr<const Fm::FileInfo> i
                         addAction(action);
                     }
                 }
-                else {
+                else if(archiver->canCreateArchive()){
```

Comment:
@tsujan the only thing changed is this line. I only show the "Compress" menu item if the archiver supports compression.
Would you like to help take a look at this to see if your archiver setting is correctly loaded?

Reply:
@tsujan  If the value of `archiver` pointer is null, then the setting is not correctly loaded.